core/src/main/scala/kafka/log/Log.scala

```diff
@@ -156,7 +157,11 @@ class Log(@volatile var dir: File,
   /* Load the log segments from the log files on disk */
   private def loadSegments() {
     // create the log directory if it doesn't exist
-    dir.mkdirs()
+    try {
+      Files.createDirectory(dir.toPath)
```

Comment:
Hmm, shouldn't this be `Files.createDirectories(dir.toPath)`? That seems to be the equivalent of `mkdirs` and it doesn't throw an exception if the directory already exists.